### PR TITLE
Dockerfile and docker-image.yml (github action)

### PR DESCRIPTION
**Pull request by pablospe**

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: 'Build and push Scantools Docker image'
```
Comment (MarioGini) [inline .github/workflows/docker-image.yml:27]: using '' here which other steps dont do

File: pipelines/pipeline_navvis_rig.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+TILE_CHOICES = sorted([attr.name.split("_")[1] for attr in TileFormat])
+
+description = """
+--------------------------------------------------------------------------------
+
+Pipeline description
+====================
+
+Converts Navvis data to Lamar/Capture format exported as rig, including meshing,
```
Comment (vjlux) [inline pipelines/pipeline_navvis_rig.py:26]: Isn't it Lamar/Kapture format. don't need to change but I want to remark its confusing

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (vjlux) [reply .github/workflows/docker-image.yml:27]: I have seen ' ' in my PR..

File: pipelines/pipeline_navvis_rig.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+**Input Parameters**
+
+    a) The input path for the data to be processed. Example:
+
+        datasets_proc/                           (--input_path "datasets_proc/")
+        ├── 2023-12-08_10.51.38
+        ├── 2023-12-15_15.45.51
+        └── 2023-12-15_15.58.10
+
+    b) The tile format. Default: "3x3".
```
Comment (vjlux) [inline pipelines/pipeline_navvis_rig.py:38]: b) Large camera images can be tiled into smaller also overlapping images. The tile format specifies the tiling algorithm. The following values are accepted: ['3x3,'5x5'], where '3x3' is the default value.

Comment (vjlux) [reply pipelines/pipeline_navvis_rig.py:38]: I feel this could be more descriptive

File: pipelines/pipeline_navvis_rig.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+    c) The output path where processed data will be saved. Defaults to the
+       current working directory. Example:
+
+        pipeline_output/                      (--output_path "pipeline_output/")
```
Comment (vjlux) [inline pipelines/pipeline_navvis_rig.py:43]: intended to be on the same line?

File: pipelines/pipeline_navvis_rig.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+        required=False,
+        default="advancing_front",
+        choices=["advancing_front", "poisson"],
+        help="Meshing method. Default: advancing_front.",
+    )
+    parser.add_argument(
+        "--meshing",
+        action=argparse.BooleanOptionalAction,
+        required=False,
+        default=True,
```
Comment (vjlux) [inline pipelines/pipeline_navvis_rig.py:264]: we always do meshing? shouldn't we do it False? meshing takes a long time, right? or is this just running the marching cube?

File: pipelines/pipeline_navvis_rig.py [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+        help="Run meshing creation. Default: True. ",
+    )
+    parser.add_argument(
+        "--rendering",
+        action=argparse.BooleanOptionalAction,
+        required=False,
+        default=True,
```
Comment (vjlux) [inline pipelines/pipeline_navvis_rig.py:271]: default=False?

File: pipelines/pipeline_navvis_rig.py [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+        help="Run depth maps rendering. Default: True. ",
+    )
+    parser.add_argument(
+        "--qrcode_detection",
+        action=argparse.BooleanOptionalAction,
+        required=False,
+        default=True,
```
Comment (vjlux) [inline pipelines/pipeline_navvis_rig.py:278]: default=False?

**Review by vjlux — APPROVED** [review]
LGTM. I left some comments though

File: pipelines/pipeline_navvis_rig.py (hunk 2 above)
Comment (pablospe) [reply pipelines/pipeline_navvis_rig.py:26]: It is Capture with 'C'. Please see this Readme: [CAPTURE.md](https://github.com/microsoft/lamar-benchmark/blob/main/CAPTURE.md)

Comment (pablospe) [reply pipelines/pipeline_navvis_rig.py:26]: Yes, it is confusing.

File: README.md [hunk 8]
```diff
@@ -87,6 +88,35 @@ python -m pip install -e .
...
+You can build the Docker images for these stages using the following commands:
+```bash
+# Build the 'scantools' stage
+docker build --target scantools -t lamar:scantools -f Dockerfile ./
+
+# Build the 'lamar' stage
+docker build --target lamar -t lamar:lamar -f Dockerfile ./
+```
+
+### Pulling the Docker Images from GitHub Docker Register
```
[the remark this answers is not in the document]
Comment (pablospe) [reply README.md:107]: Not here. This is for building the docker image yourself. The register version is in the CI:
https://github.com/microsoft/lamar-benchmark/pull/57/files/95c0c4565f3f855c77255d09fb138636c35d3f22#diff-dbade1e5797a7633e024a7685a43a7cbd96ae6fefc6314294807ae34cd83712cR34

File: README.md [hunk 9]
```diff
@@ -87,6 +88,35 @@ python -m pip install -e .
...
+
+Alternatively, if you don't want to build the images yourself, you can pull them
+from the GitHub Docker Register using the following commands:
```
[the remark this answers is not in the document]
Comment (pablospe) [reply README.md:110]: apologies, what did you mean?

File: pipelines/pipeline_navvis_rig.py (hunk 5 above)
Comment (pablospe) [reply pipelines/pipeline_navvis_rig.py:264]: This is a full pipeline, make senses to select the defaults we want to have. The individual scripts can be called separately, like: `scantools.run_navvis_to_capture.py`

File: pipelines/pipeline_navvis_rig.py (hunk 4 above)
Comment (pablospe) [reply pipelines/pipeline_navvis_rig.py:45]: yes, this is what it needs to be in the command line. It is understandable when one does `pipelines/pipeline_navvis_rig.py --help`

**Review by vjlux — APPROVED** [review]

**Review by vjlux — APPROVED** [review]

**Review by mihaidusmanu — APPROVED** [review]

**Pull request merged**